### PR TITLE
feature/SDK-41_oidf-support

### PR DESCRIPTION
**Pull request by sanderPostma**

### CODE REVIEW

File: packages/siopv2-oid4vp-rp-auth/src/types/ISIOPv2RP.ts [hunk 1]
```diff
@@ -193,6 +195,7 @@ export interface IPresentationWithDefinition {
 export interface ISIOPIdentifierOptions extends Omit<IDIDOptions, 'idOpts'> {
   // we replace the legacy idOpts with the Managed Identifier opts from the identifier resolution module
   idOpts: ManagedIdentifierOptsOrResult
+  oidfOpts?: ExternalIdentifierOIDFEntityIdOpts
```
Comment (sanderPostma) [inline packages/siopv2-oid4vp-rp-auth/src/types/ISIOPv2RP.ts:198]: Took a shortcut here by keeping it side-by-side

File: packages/oidf-client/package.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
 	"name": "@sphereon/ssi-sdk.oidf-client",
+	"type": "module",
```
Comment (BtencateSphereon) [inline packages/oidf-client/package.json:3]: hmm is this a change we should make?

File: packages/tsconfig.json [hunk 3]
```diff
@@ -48,6 +48,6 @@
     { "path": "ebsi-support" },
     { "path": "public-key-hosting" },
     { "path": "resource-resolver" },
-	{ "path":  "oidf-client" }
+	{ "path": "oidf-client" }
```
Comment (BtencateSphereon) [inline packages/tsconfig.json:51]: missing oidf-metatdata-server?

File: packages/kv-store/src/keyv/keyv.ts [hunk 4]
```diff
@@ -156,6 +158,39 @@ export class Keyv<Value = any> extends EventEmitter implements KeyvStore<Value>
       .then(() => Promise.all(results))
   }
 
+  private isMapWithEntries(store: any): store is Map<string, Value> {
+    return store instanceof Map && typeof store.entries === 'function'
+  }
+
+  private async getFromMapWithPrefix(
+    prefixes: Array<string>,
+    options?: {
```
Comment (BtencateSphereon) [inline packages/kv-store/src/keyv/keyv.ts:167]: should we use this?

File: packages/kv-store/src/keyv/keyv.ts [hunk 5]
```diff
@@ -156,6 +158,39 @@ export class Keyv<Value = any> extends EventEmitter implements KeyvStore<Value>
...
+  ): Promise<Array<KeyvStoredData<Value>>> {
+    if (!this.isMapWithEntries(this.store)) {
+      return []
+    }
+
+    const map = this.store as Map<string, Value>
+    const entries = Array.from(map.entries())
+
+    const result = []
+    for (const prefix of prefixes) {
```
Comment (BtencateSphereon) [inline packages/kv-store/src/keyv/keyv.ts:179]: can we not clean this stuff up with a reduce?

something like this

const map = this.store as Map<string, Value>;
const entries = Array.from(map.entries());

const result = prefixes.reduce((acc, prefix) => {
  const matchingEntry = entries.find(([key]) => key.startsWith(prefix));
  if (matchingEntry) {
    acc.push(matchingEntry[1] as KeyvStoredData<Value>);
  } else {
    acc.push(undefined);
  }
  return acc;
}, []);

return result;

File: packages/oid4vci-issuer-store/src/agent/OID4VCIStore.ts [hunk 6]
```diff
@@ -84,17 +90,19 @@ export class OID4VCIStore implements IAgentPlugin {
         }),
       )
     }
+
     if (opts && Array.isArray(opts?.importMetadatas)) {
-      opts.importMetadatas.forEach((meta) =>
-        this.oid4vciStorePersistMetadata({
+      opts.importMetadatas.forEach((metaImport: IMetadataImportArgs) => {
+        const meta = metaImport as IIssuerMetadataImportArgs
```
Comment (BtencateSphereon) [inline packages/oid4vci-issuer-store/src/agent/OID4VCIStore.ts:96]: is this cast safe?

**Review by BtencateSphereon — CHANGES_REQUESTED** [review]

File: packages/oidf-client/package.json (hunk 2 above)
Comment (sanderPostma) [reply packages/oidf-client/package.json:3]: No idea, it's from John's branch SDK-44 which I needed for the new oidf-client. He must have added it for a reason

File: packages/tsconfig.json (hunk 3 above)
Comment (sanderPostma) [reply packages/tsconfig.json:51]: fixed

File: packages/kv-store/src/keyv/keyv.ts (hunk 4 above)
Comment (sanderPostma) [reply packages/kv-store/src/keyv/keyv.ts:167]: removed, this started as a copy from get()

File: packages/oidf-client/package.json (hunk 2 above)
Comment (BtencateSphereon) [reply packages/oidf-client/package.json:3]: so this PR is now merging in SDK-44? this is not what we should do then? this branch should then have been based on SDK-44?

Comment (BtencateSphereon) [reply packages/oidf-client/package.json:3]: also setting type to module specifies the type of modules that the project will use. and this is the only one with this option. and from my experience is most likely never the solution to the problem

Comment (sanderPostma) [reply packages/oidf-client/package.json]: Fixed, SDK-44 is merged also, then I synched up with develop

**Review by BtencateSphereon — APPROVED** [review]

**Pull request merged**